pkg/asset/manifests/cloudproviderconfig.go

```diff
@@ -90,6 +96,12 @@ func (cpc *CloudProviderConfig) Generate(dependencies asset.Parents) error {
 			return err
 		}
 		cm.Data[cloudProviderConfigDataKey] = string(marshalled)
+	case azuretypes.Name:
+		azureConfig, err := azuremanifests.CloudProviderConfig(installConfig.Config.Platform.Azure, clusterID.InfraID)
+		if err != nil {
+			return errors.Wrap(err, "could not create cloud provider config")
+		}
+		cm.Data[cloudProviderConfigDataKey] = azureConfig
```

Comment:
as you see at https://github.com/openshift/installer/pull/1606/files#diff-8db489d59cedb0a4f04c7546779f58a2R68 you are writing secret credentials to a configmap and that is not allowed.

vsphere other platform that supports creating cloudconfig uses secret to fetch the cloud credentials. This needs more thought.